frontend/src/ui/InitialConsent.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef } from "react";
+import CONFIG from "../config";
+import { currentRef, useTranslatedConfig } from "../util";
+import { Modal, ModalHandle } from "./Modal";
+import { Button } from "./Button";
+import { TextBlock } from "./Blocks/Text";
+
+
+export const InitialConsent: React.FC = () => {
+    const modalRef = useRef<ModalHandle>(null);
+    const userConsent = localStorage.getItem("userConsent") ?? "";
+    const { title, text, button } = CONFIG.initialConsent;
+
+    // Since this doesn't store any critical information, a simple, insecure hash
+    // should suffice.
+    // Source: https://gist.github.com/jlevy/c246006675becc446360a798e2b2d781
+    const simpleHash = (str: string) => {
+        let hash = 0;
+        for (let i = 0; i < str.length; i++) {
+            hash = ((hash << 5) - hash + str.charCodeAt(i)) | 0;
+        }
+        return (hash >>> 0).toString(36);
+    };
+
+    const hash = simpleHash(CONFIG.initialConsent.toString());
+
+    return userConsent !== hash ? (
+        <Modal
+            open
+            ref={modalRef}
+            title={useTranslatedConfig(title)}
+            closable={false}
+        >
+            <TextBlock content={useTranslatedConfig(text)} />
+            <Button autoFocus css={{ marginTop: 20 }} onClick={() => {
```

Comment:
otherwise it would focus the first link, which is also kind of weird (that is the default behaviour of the focus-trap thing the modal uses). But I think it's also possible to not focus anything instead. I might even prefer that.

Reply:
Ah ye, focusing nothing would probably be preferred. But again, not too important, if it's too hard, its fine focusing the button.